### PR TITLE
Updated configmap_controller_test.go

### PR DESCRIPTION
**Pull request by alejandroEsc**
**What this PR does / why we need it**: A few style decisions to make this test a bit harder to break. Including a fix to help it fail and not panic when channel produces a nil.

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by nikhiljindal** [comment]
CC @csbell to review

**Comment by csbell** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[federation/pkg/federation-controller/configmap/configmap_controller_test.go, line 167 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562#-KdBd0Gczhante6sMdod:-KdBd0Gczhante6sMdoe:bwgwqej) ([raw file](https://github.com/kubernetes/kubernetes/blob/97a2f24c01270ca7e8a5c9627144abe891e449ad/federation/pkg/federation-controller/configmap/configmap_controller_test.go#L167)):*
> ```Go
> 		return nil
> 	} else {
> 		return configmap.(*apiv1.ConfigMap)
> ```

Thanks. The flake surfaced the above deref problem that occurs in other test controllers (I can see at least secrets and namespaces). Can you address the issue there as well?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562)*

**Comment by alejandroEsc** [comment]
*[federation/pkg/federation-controller/configmap/configmap_controller_test.go, line 167 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562#-KdBd0Gczhante6sMdod:-KdBgdmX3cc7yOgH7ZNk:b-ou7n8e) ([raw file](https://github.com/kubernetes/kubernetes/blob/97a2f24c01270ca7e8a5c9627144abe891e449ad/federation/pkg/federation-controller/configmap/configmap_controller_test.go#L167)):*
<details><summary><i>Previously, csbell (Christian Bell) wrote…</i></summary><blockquote>

Thanks. The flake surfaced the above deref problem that occurs in other test controllers (I can see at least secrets and namespaces). Can you address the issue there as well?
</blockquote></details>

Sounds good, Ill try to make the same change where I find it.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562)*

**Comment by alejandroEsc** [comment]
@csbell made the requested changes and ready for you to give a look.

**Comment by alejandroEsc** [comment]
@k8s-bot cvm gke e2e test this
@k8s-bot gci gke e2e test this

**Comment by alejandroEsc** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by alejandroEsc** [comment]
@k8s-bot gci gke e2e test this

**Comment by alejandroEsc** [comment]
@k8s-bot gci gce e2e test this

**Comment by alejandroEsc** [comment]
@k8s-bot gci gce e2e test this

**Comment by alejandroEsc** [comment]
@k8s-bot cvm gke e2e test this

**Comment by alejandroEsc** [comment]
@k8s-bot non-cri e2e test this
@k8s-bot gci gke e2e test this

**Comment by alejandroEsc** [comment]
@k8s-bot gci gce e2e test this

**Comment by alejandroEsc** [comment]
@csbell sorry about the test spam, seems like i got caught up on a lot of flakes there. Anyway things are looking good, please have another look, thank you.

**Comment by alejandroEsc** [comment]
Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[federation/pkg/federation-controller/configmap/configmap_controller_test.go, line 167 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562#-KdBd0Gczhante6sMdod:-KdbgE5vnzCtjidAY8o2:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/97a2f24c01270ca7e8a5c9627144abe891e449ad/federation/pkg/federation-controller/configmap/configmap_controller_test.go#L167)):*
<details><summary><i>Previously, alejandroEsc (Alejandro Escobar) wrote…</i></summary><blockquote>

Sounds good, Ill try to make the same change where I find it.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562)*

File: federation/pkg/federation-controller/daemonset/daemonset_controller_test.go [hunk 1]
```diff
@@ -38,27 +38,33 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const (
+	daemonsets string = "daemonsets"
+	clusters   string = "clusters"
+	errorMsgF  string = "expected: %v, actual: %v"
```
Comment (csbell) [inline federation/pkg/federation-controller/daemonset/daemonset_controller_test.go]: const printf-style format strings make them less readable in the code where they are used.

File: federation/pkg/federation-controller/namespace/namespace_controller_test.go [hunk 2]
```diff
@@ -39,6 +39,11 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const (
+	namespaces        = "namespaces"
```
Comment (csbell) [inline federation/pkg/federation-controller/namespace/namespace_controller_test.go]: namespaces string = "namespaces" for consistency.

File: federation/pkg/federation-controller/secret/secret_controller_test.go [hunk 3]
```diff
@@ -39,27 +39,33 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const (
+	clusters  string = "clusters"
+	secrets   string = "secrets"
+	errorMsgF string = "expected: %v, actual: %v"
```
Comment (csbell) [inline federation/pkg/federation-controller/secret/secret_controller_test.go]: Ditto about format strings.

**Review by csbell — COMMENTED** [review]
Small nits

**Comment by alejandroEsc** [comment]
@csbell nit items addressed, thank you for reviewing!

**Comment by alejandroEsc** [comment]
Review status: 0 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

*[federation/pkg/federation-controller/configmap/configmap_controller_test.go, line 167 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562#-KdBd0Gczhante6sMdod:-Kdh5UtLJxDkgRw00DuK:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/97a2f24c01270ca7e8a5c9627144abe891e449ad/federation/pkg/federation-controller/configmap/configmap_controller_test.go#L167)):*
<details><summary><i>Previously, alejandroEsc (Alejandro Escobar) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[federation/pkg/federation-controller/daemonset/daemonset_controller_test.go, line 44 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562#-KdXHUMJzMmuze7u5-_s-r2-44:-Kdh5NXHOUWcEZ2IBHgb:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/2519039388b2ac6afca701363531d6f50d6b422c/federation/pkg/federation-controller/daemonset/daemonset_controller_test.go#L44)):*
<details><summary><i>Previously, csbell (Christian Bell) wrote…</i></summary><blockquote>

const printf-style format strings make them less readable in the code where they are used.
</blockquote></details>

Done.

---

*[federation/pkg/federation-controller/namespace/namespace_controller_test.go, line 43 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562#-KdXHUMJzMmuze7u5-_v-r2-43:-Kdh5OAgS9kzvsxKXgni:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/2519039388b2ac6afca701363531d6f50d6b422c/federation/pkg/federation-controller/namespace/namespace_controller_test.go#L43)):*
<details><summary><i>Previously, csbell (Christian Bell) wrote…</i></summary><blockquote>

namespaces string = "namespaces" for consistency.
</blockquote></details>

Done.

---

*[federation/pkg/federation-controller/secret/secret_controller_test.go, line 45 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562#-KdXHUMJzMmuze7u5-_x-r2-45:-Kdh5SWnemhTrf335gW6:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/2519039388b2ac6afca701363531d6f50d6b422c/federation/pkg/federation-controller/secret/secret_controller_test.go#L45)):*
<details><summary><i>Previously, csbell (Christian Bell) wrote…</i></summary><blockquote>

Ditto about format strings.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562)*

**Comment by alejandroEsc** [comment]
@csbell do you need to close out reviewable? It seems to be the one pending check, sorry I am not a reviewable expert here.

**Comment by csbell** [comment]
Reviewed 4 of 7 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/41562)*

**Comment by csbell** [comment]
@alejandroEsc Indeed. Should be good now! Thanks for fixing up the other tests.

**Pull request merged**